recipes/zlib-ng/all/conanfile.py

```diff
@@ -51,35 +46,34 @@ def configure(self):
         del self.settings.compiler.cppstd
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
+        files.get(self, **self.conan_data["sources"][self.version],
             destination=self._source_subfolder, strip_root=True)
 
     def validate(self):
-        if self.options.zlib_compat and not self.options.with_gzfileop:
+        if self.info.options.zlib_compat and not self.info.options.with_gzfileop:
```

Comment:
is it valid even in conan v1 mode?

Reply:
Yes, it does. The only case I still have no idea how fix, is when a package can be header-only or not. When the package id sanitized, self.info.options.xxx and self.info.setting.xxx are None, so you can't validate and there is no get_safe available. I've opened an issue for more information: https://github.com/conan-io/conan/issues/11786

Reply:
I was told there will be new method `validate_build` to address that (doesn't look promising...)
right now
this works in v2, but fails in v1:
```
cppstd = self.info.settings.get_safe("compiler.cppstd")
```
this works in v1, but fails in v2:
```
cppstd = self.settings.get_safe("compiler.cppstd")
```
the only way I have found to satisfy both:
```
if self.info.settings.get_safe is not None:
    cppstd = self.info.settings.get_safe("compiler.cppstd")
else:
    cppstd = self.settings.get_safe("compiler.cppstd")
```
(even `hasattr` didn't work as expected 🤔 )

Reply:
This recipe is correct about the `validate` as it is checking if a binary is possible or not.
I responded to the issue opened by @uilianries, I hope now it makes sense: https://github.com/conan-io/conan/issues/11786

Reply:
@SSE4 Please let us know, in the conan developer channel why you consider this feature `doesn't look promising` or open an issue as @uilianries did. I don't think that sentence is the optimal way to express doubts about a feature that your team designed.